src/osd/ECBackend.cc

```diff
@@ -1100,6 +1100,13 @@ void ECBackend::handle_sub_read(
   }
   reply->from = get_parent()->whoami_shard();
   reply->tid = op.tid;
+  if (from == get_parent()->whoami_shard()) {
```

Comment:
would it be simpler to do this in the caller?  that way teh caller declares reply on teh stack, processes the message to create the reply, and then processes the reply.  and this method would remain unchanged...